src/components/Api.js

```diff
@@ -0,0 +1,152 @@
+import { token } from "../utils/constants.js";
+
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  editUserInfo({ name, job }) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: { ...this._headers, "Content-Type": "application/json" },
```

Comment:
According to the task you need to pass full `headers` into the constructor of `Api`. See here https://snipboard.io/pRdaAZ.jpg . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)